src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,59 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    if (this.constructor.name !== 'Herbivore') {
+      Animal.alive.push(this);
+    }
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+
+    Animal.alive = Animal.alive.map((objAnimal) => {
+      if (objAnimal.name === this.name) {
+        return { ...objAnimal, hidden: !objAnimal.hidden };
+      } else {
+        return objAnimal;
+      }
+    });
```

Comment:
`Animal.alive` holds a reference to the animal, every change done to the object will already be accessible through `Animal.alive`; there is no need for this map!